index.js

```diff
@@ -96,8 +96,7 @@ FibonacciHeap.prototype.extractMinimum = function () {
 
     // Merge the children of the minimum node with the root list
     this.minNode = mergeLists(nextInRootList, extractedMin.child, this.compare);
-    if (nextInRootList) {
-      this.minNode = nextInRootList;
+    if (this.minNode) {
```

Comment:
No, it's not an optimization.  
If you do not consolidate after extracting for some reason - you are setting wrong minNode pointer in most cases. Since nextInRootList is not guaranteed to be smallest among root list nodes.